File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/entity/component/PositionComponentTest.java
package net.bestia.zoneserver.entity.component;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import net.bestia.model.domain.Direction;
import net.bestia.model.geometry.Point;

public class PositionComponentTest {

	private PositionComponent posComp;

	@Before
	public void setup() {
		posComp = new PositionComponent(10, 12);
	}

	@Test
	public void isSightBlocking_setAndGet() {
		Assert.assertFalse(posComp.isSightBlocking());
		posComp.setSightBlocking(true);
		Assert.assertTrue(posComp.isSightBlocking());
	}

	@Test
	public void getFacing_setAndGet() {
		posComp.setFacing(Direction.EAST);
		Assert.assertEquals(Direction.EAST, posComp.getFacing());
	}
	
	@Test
	public void setPosition_xAndY() {
		posComp.setPosition(123, 69);
		Assert.assertEquals(123, posComp.getPosition().getX());
		Assert.assertEquals(69, posComp.getPosition().getY());
	}
	
	@Test
	public void setPosition_point() {
		Point p = new Point(12, 14);
		posComp.setPosition(p);
		Assert.assertEquals(p, posComp.getPosition());
	}
}

File: bestia-zoneserver/src/test/java/net/bestia/entity/component/PositionComponentTest.kt
package net.bestia.entity.component

import net.bestia.model.domain.Direction
import net.bestia.model.geometry.Point
import net.bestia.zoneserver.entity.component.PositionComponent
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class PositionComponentTest {

  private lateinit var posComp: PositionComponent

  @Before
  fun setup() {
    posComp = PositionComponent(10)
  }

  @Test
  fun isSightBlocking_setAndGet() {
    Assert.assertFalse(posComp!!.isSightBlocking)
    posComp.isSightBlocking = true
    Assert.assertTrue(posComp!!.isSightBlocking)
  }

  @Test
  fun getFacing_setAndGet() {
    posComp.facing = Direction.EAST
    Assert.assertEquals(Direction.EAST, posComp!!.facing)
  }

  @Test
  fun setPosition_xAndY() {
    posComp.position = Point(123, 69)
    Assert.assertEquals(123, posComp.position.x)
    Assert.assertEquals(69, posComp.position.y)
  }

  @Test
  fun setPosition_point() {
    val p = Point(12, 14)
    posComp.position = p
    Assert.assertEquals(p, posComp!!.position)
  }
}
